qa-deploy

```diff
@@ -0,0 +1,65 @@
+#! /usr/bin/env bash
+
+# Bash strict mode
+set -euo pipefail
+
+USAGE="Usage
+===
+
+  $ ./qa-deploy PRODUCTION_URL [STAGING_URL]
+
+Description
+---
+Deploy locally to minikube.
+"
+
+function invalid() {
+    message=${1}
+    echo "Error: ${message}"
+    echo ""
+    echo "$USAGE"
+    exit 1
+}
+
+function run_minikube() {
+    # Run minikube
+    if ! [ minikube ip 2> /dev/null > /dev/null ]; then
+        minikube start
+    fi
+
+    minikube addons enable ingress
+    kubectl config set-context minikube
+}
+
+function set_up_environment() {
+    kubectl apply -f qa/config-maps/proxy.config.yaml
+}
+
+function deploy_service() {
+    service="${1}"
+
+    cat "services/${service}.yaml" | sed 's|prod-comms.docker-registry.canonical.com|canonicalwebteam|' | sed 's|[:][$][{]TAG_TO_DEPLOY[_aa-zA-Z]*[}]||' | sed 's|replicas: [0-9][0-9]*|replicas: 1|' | kubectl apply --filename -
+}
+
+function deploy_ingress() {
+    ingress="${1:-}"
```

Comment:
This falls back to a default value but we don't catch if it isn't set, even though it is required.
We could remove the `:-` for now.